plus/provider/src/main/java/org/sakaiproject/plus/ProviderServlet.java

```diff
@@ -0,0 +1,1756 @@
+
+	@Autowired private ServerConfigurationService serverConfigurationService;
+	@Autowired private SiteMembershipUpdater siteMembershipUpdater;
+	@Autowired private SiteMembershipsSynchroniser siteMembershipsSynchroniser;
+	@Autowired private SiteEmailPreferenceSetter siteEmailPreferenceSetter;
+	@Autowired private UserFinderOrCreator userFinderOrCreator;
+	@Autowired private UserLocaleSetter userLocaleSetter;
+	@Autowired private UserPictureSetter userPictureSetter;
+	@Autowired private LTIService ltiService;
+	@Autowired private PlusService plusService;
+	@Autowired private TenantRepository tenantRepository;
+	@Autowired private ContextRepository contextRepository;
+
+	private List<BLTIProcessor> bltiProcessors = new ArrayList();
+
+	private String randomUUID = UUID.randomUUID().toString();
+
+	private KeyPair localKeyPair = LTI13Util.generateKeyPair();
+
+	private static Cache getCache()
+	{
+		CacheManager cacheManager = (CacheManager) ComponentManager.get("org.sakaiproject.ignite.SakaiCacheManager");
+		return cacheManager.getCache(CACHE_NAME);
+	}
+
+	public static String getCacheKey(String key)
+	{
+		Cache cache = getCache();
+		Cache.ValueWrapper value = cache.get(key);
+		if ( value == null ) return null;
+		return (String) value.get();
+	}
+
+	public static void putCacheKey(String key, String value)
+	{
+		Cache cache = getCache();
+		cache.put(key, value);
+	}
+
+	/**
+	 * Setup a security advisor.
+	 */
+	public void pushAdvisor() {
+		// setup a security advisor
+		SecurityService.pushAdvisor(new SecurityAdvisor() {
+			public SecurityAdvice isAllowed(String userId, String function,
+					String reference) {
+				return SecurityAdvice.ALLOWED;
+			}
+		});
+	}
+
+	/**
+	 * Remove our security advisor.
+	 */
+	public void popAdvisor() {
+		SecurityService.popAdvisor();
+	}
+
+	// TODO: Make this a *lot* prettier and add forward to knowledge base feature :)
+	public void doError(HttpServletRequest request,HttpServletResponse response, String s, String message, Throwable e) throws java.io.IOException {
+		if (e != null) {
+			log.error(e.getLocalizedMessage(), e);
+		}
+		log.info("{}: {}", rb.getString(s), message);
+		PrintWriter out = response.getWriter();
+		out.println(rb.getString(s));
+		out.println(htmlEscape(message));
+		if ( e != null ) out.println(e.getMessage());
+	}
+
+	// TODO: Make this a *lot* prettier and add forward to knowledge base feature :)
+	public void addError(PrintWriter out, String s, String message, Throwable e) throws java.io.IOException {
+		if (e != null) {
+			log.error(e.getLocalizedMessage(), e);
+		}
+		log.info("{}: {}", rb.getString(s), message);
+		out.println(rb.getString(s));
+		out.println(htmlEscape(message));
+		if ( e != null ) out.println(e.getMessage());
+	}
+
+	@Override
+	public void init(ServletConfig config) throws ServletException {
+
+		super.init(config);
+		SpringBeanAutowiringSupport.processInjectionBasedOnCurrentContext(this);
```

Comment:
this should likely go in the constructor vs the init

Reply:
You have it in the init() in the file I borrowed the pattern from:

lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/ccexport/CCExportServlet.java

While I am not an expert - I would think that the init() is something that Spring has more control over than the constructor - I don't think that Spring can cause the constructor to run twicwe but it can call init() twice.

But I have no clue :)   I just did it the same way you did.